test/tests/pbs_smoketest.py

```diff
@@ -1321,19 +1321,25 @@ def test_resource_delete(self):
                             self.resc_name, t, f, c, k, v)
                         self.logger.info("")
 
-    def setup_fs(self):
-
-        self.scheduler.set_sched_config({'log_filter': '0'})
-
-        self.scheduler.add_to_resource_group('grp1', 100, 'root', 60)
-        self.scheduler.add_to_resource_group('grp2', 200, 'root', 40)
-        self.scheduler.add_to_resource_group('pbsuser1', 101, 'grp1', 40)
-        self.scheduler.add_to_resource_group('pbsuser2', 102, 'grp1', 20)
-        self.scheduler.add_to_resource_group('pbsuser3', 201, 'grp2', 30)
-        self.scheduler.add_to_resource_group('pbsuser4', 202, 'grp2', 10)
+    def setup_fs(self, formula):
+
+        # change resource group file and validate after all the changes are in
+        self.scheduler.add_to_resource_group('grp1', 100, 'root', 60,
+                                             validate=False)
+        self.scheduler.add_to_resource_group('grp2', 200, 'root', 40,
+                                             validate=False)
+        self.scheduler.add_to_resource_group('pbsuser1', 101, 'grp1', 40,
+                                             validate=False)
+        self.scheduler.add_to_resource_group('pbsuser2', 102, 'grp1', 20,
+                                             validate=False)
+        self.scheduler.add_to_resource_group('pbsuser3', 201, 'grp2', 30,
+                                             validate=False)
+        self.scheduler.add_to_resource_group('pbsuser4', 202, 'grp2', 10,
```

Comment:
I think, validate does not validate any specific data that is being added to the resource group file. There is actually no way for it to do that. It just validates that on scheduler HUP the configuration was read by scheduler without reporting any error. The scheduler does report a line number associated to the configuration which it could not parse. So, ideally just one validate at the end should just validate everything else present in the resource group file.

Even if I add apply argument, I will still have to pass validate as false to the call because it takes validate=True by default. Or, I'll have to do some check wherein if apply is set then do not use validate or something like that. 

IMO a better way to fix this problem is to allow a list of strings (or some objects containing resource group elements) that we wish to add to the resource group file and validate it once. But, I am not trying to fix the interface here, my purpose was to get rid of timeout I got in Travis run.

So, if you do not feel strongly about the change you are suggesting then I'd like to keep it as is.